### PR TITLE
OSSM-5398 Add test checking spec.values validation

### PR DESCRIPTION
**Pull request by bmangoen**

### CODE REVIEW

File: controllers/istio_controller_test.go [hunk 1]
```diff
@@ -78,10 +105,7 @@ var _ = Describe("IstioController", Ordered, func() {
 				},
 				Spec: v1.IstioSpec{
 					Version: istioVersion,
-					Values: []byte(`{
-						"pilot":{"image":"` + pilotImage + `"},
-						"istio_cni":{"enabled":true}
-					}`),
+					Values:  istioValues,
```
Comment (luksa) [inline controllers/istio_controller_test.go:108]: Not sure if it's a good idea to change the values of the existing test. Better to add a new test.

File: controllers/istio_controller_test.go [hunk 2]
```diff
@@ -50,6 +50,33 @@ var _ = Describe("IstioController", Ordered, func() {
...
+		}
+	}`)
+
+	istioCNIValues := string(`{
+		"enabled":true,
+		"excludeNamespaces": ["openshift", "kube-system"]
+	}`)
+
+	istioValues := []byte(`{
+		"donotexist": "shouldbeomitted",
```
Comment (luksa) [inline controllers/istio_controller_test.go:74]: This value should throw an error instead of being silently ignored.

Comment (bmangoen) [reply controllers/istio_controller_test.go:74]: Actually, this value is omitted without throwing an error by the controller-runtime client but only as an info warning `INFO    KubeAPIWarningLogger    unknown field "spec.values.donotexist"`. Should we try to catch this as an error?

**Pull request closed without merging**